buttons/embed.py

```diff
@@ -98,51 +100,46 @@ def embed(self):
-
     def add_page_numbers(self):
         """Set footers with page numbers for each embed in list"""
         for page, embed in enumerate(self.embeds):
             utils.embed.add_author_footer(
                 embed, self.author, additional_text=[f"Page {page+1}/{self.max_page}"]
             )
 
+    async def lock_callback(self, interaction: disnake.MessageInteraction):
+        self.dynam_lock = not self.dynam_lock
+        if self.dynam_lock:
+            self.lock_button.style = disnake.ButtonStyle.danger
+            self.lock_button.emoji = "🔒"
+        else:
+            self.lock_button.style = disnake.ButtonStyle.success
+            self.lock_button.emoji = "🔓"
+        await interaction.response.edit_message(view=self)
+
+    async def start_callback(self, interaction: disnake.MessageInteraction):
+        self.page = utils.embed.pagination_next("start", self.page, self.max_page, self.roll_around)
+        await interaction.response.edit_message(embed=self.embed, view=self)
+
+    async def prev_callback(self, interaction: disnake.MessageInteraction):
+        self.page = utils.embed.pagination_next("prev", self.page, self.max_page, self.roll_around)
+        await interaction.response.edit_message(embed=self.embed, view=self)
+
+    async def next_callback(self, interaction: disnake.MessageInteraction):
+        self.page = utils.embed.pagination_next("next", self.page, self.max_page, self.roll_around)
+        await interaction.response.edit_message(embed=self.embed, view=self)
+
+    async def end_callback(self, interaction: disnake.MessageInteraction):
+        self.page = utils.embed.pagination_next("end", self.page, self.max_page, self.roll_around)
+        await interaction.response.edit_message(embed=self.embed, view=self)
+
     async def interaction_check(self, interaction: disnake.MessageInteraction) -> bool:
-        if interaction.data.custom_id == "embed:lock":
-            if interaction.author.id == self.author.id:
-                self.locked = not self.locked
-                if self.locked:
-                    self.lock_button.style = disnake.ButtonStyle.danger
-                    self.lock_button.emoji = "🔒"
-                else:
-                    self.lock_button.style = disnake.ButtonStyle.success
-                    self.lock_button.emoji = "🔓"
-                await interaction.response.edit_message(view=self)
-            else:
-                await interaction.send(Messages.embed_not_author, ephemeral=True)
-            return False
-        ids = ["embed:start_page", "embed:prev_page", "embed:next_page", "embed:end_page"]
-        if interaction.data.custom_id not in ids or self.max_page <= 1:
-            return False
-        if (self.perma_lock or self.locked) and interaction.author.id != self.author.id:
+        if (self.perma_lock and interaction.author.id != self.author.id) or (
+            self.dynam_lock and interaction.author.id != self.author.id
+        ):
```

Comment:
This is the same condition but shorter. It is also how it looked originally, not sure why you made it more complicated.

```suggestion
        if (self.perma_lock or self.dynam_lock) and interaction.author.id != self.author.id):
```